halo2_gadgets/src/poseidon.rs

```diff
@@ -180,7 +171,8 @@ impl<
             &mut self.state,
             Some(&self.mode),
         )?;
-        self.mode = Absorbing::init_with(value);
+        self.mode = Absorbing::init_empty();
+        self.mode.absorb(value).expect("state is not full");
```

Comment:
Doesn't this suggest that https://github.com/zcash/halo2/pull/831/commits/0d883685d191e019915db41c7c2124f09e0f1464#diff-76142ee2fee95e1ba7ebdff8392dfb7442517333aaf489d5cf92d6dcebee37a4R190-R194 should be public?